velox/type/StringToFloatParser.cpp

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/type/StringToFloatParser.h"
+
+namespace facebook::velox::util {
+
+static inline bool characterIsSpace(char c) {
+  return c == ' ' || c == '\t' || c == '\n' || c == '\v' || c == '\f' ||
+      c == '\r';
+}
+
+static inline bool
+isEither(const char* str, int offset, char lower, char upper) {
+  char ch = *(str + offset);
+  return ch == lower || ch == upper;
+}
+
+static inline bool isInfinityString(const char* str, size_t length) {
+  // length = 3: inf/INF
+  // length = 8: infinity
+  if (length != 3 && length != 8) {
+    return false;
+  }
+
+  if (isEither(str, 0, 'i', 'I')) {
```

Comment:
wondering if it would be more readable as: 

```
if (isEither(str[0], 'i', 'I') && isEither(str[0], 'n', 'N') && isEither(str[0], 'f', 'F')) {
   if (length == 3) {
       return true;
   }
   
   return isEither(str[3], 'i', 'I') && ...;
}
```

Reply:
let me refactor it.